### PR TITLE
Methods for default sequence dictionary name

### PR DESCRIPTION
**Pull request by magicDGS**
### Description

Changes for https://github.com/broadinstitute/picard/pull/712 and https://github.com/broadinstitute/gatk/pull/2243

### Checklist

- [x] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

**Comment by magicDGS** [comment]
This is for two PRs under review in Picard and GATK. @yfarjoun and @cmnbroad, you are reviewing them...could you have a look?

File: src/main/java/htsjdk/samtools/reference/ReferenceSequenceFileFactory.java [hunk 1]
```diff
@@ -113,24 +115,53 @@ public static ReferenceSequenceFile getReferenceSequenceFile(final Path path, fi
...
 
-        throw new IllegalArgumentException("File is not a supported reference file type: " + path.toAbsolutePath());
+    /**
+     * Returns the FASTA extension for the path.
+     *
+     * @param path the reference sequence file path.
+     *
+     * @throws IllegalArgumentException if the file is not a supported reference file.
+     */
+    public static String getFastaExtension(final Path path) {
```
Comment (magicDGS) [inline src/main/java/htsjdk/samtools/reference/ReferenceSequenceFileFactory.java]: I don't know about this, maybe it is better to discuss it in #724...

File: src/test/java/htsjdk/samtools/reference/ReferenceSequenceFileFactoryTests.java [hunk 2]
```diff
@@ -36,4 +37,22 @@
...
+                {"break.txt.gz.txt.gz", "break.txt.gz.dict"},
+                {"break.fasta.gz.fasta.gz", "break.fasta.gz.dict"}
+        };
+    }
+
+    @Test(dataProvider = "fastaNames")
+    public void testGetDefaultDictionaryForReferenceSequence(final String fastaFile, final String expectedDict) throws Exception {
+        Assert.assertEquals(ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(new File(fastaFile)), new File(expectedDict));
+    }
+
```
Comment (magicDGS) [inline src/test/java/htsjdk/samtools/reference/ReferenceSequenceFileFactoryTests.java]: Removed

File: src/main/java/htsjdk/samtools/reference/ReferenceSequenceFileFactory.java [hunk 3]
```diff
@@ -113,24 +115,53 @@ public static ReferenceSequenceFile getReferenceSequenceFile(final Path path, fi
...
+        getFastaExtension(path);
+        // Using faidx requires truncateNamesAtWhitespace
+        if (truncateNamesAtWhitespace && preferIndexed && IndexedFastaSequenceFile.canCreateIndexedFastaReader(path)) {
+            try {
+                return new IndexedFastaSequenceFile(path);
+            }
+            catch (final FileNotFoundException e) {
+                throw new IllegalStateException("Should never happen, because existence of files has been checked.", e);
             }
         }
```
Comment (magicDGS) [inline src/main/java/htsjdk/samtools/reference/ReferenceSequenceFileFactory.java]: That was out of the PR, but fixed.

**Comment by magicDGS** [comment]
Adreessed requests. Back to you @yfarjoun.

**Comment by yfarjoun** [comment]
thanks!

**Comment by yfarjoun** [comment]
👍

**Pull request merged**